app/models/fe/page.rb

```diff
@@ -53,7 +53,11 @@ class Page < ActiveRecord::Base
     # end
 
     def has_questions?
-      questions.present? || question_grids.present? || question_grid_with_totals.present?
+      all_questions.present?
+    end
+
+    def all_questions
+      all_elements.where("kind NOT IN('Fe::Paragraph', 'Fe::Section', 'Fe::QuestionGrid', 'Fe::QuestionGridWithTotal')")
```

Comment:
Seeing it in the scope and here is what made me suggest extracting it. If the scope is the only place left in the code that it shows up, that's fine.